tests/console/yast2_dns_server.pm

```diff
@@ -25,12 +25,19 @@ use version_utils qw(is_leap is_sle sle_version_at_least);
 sub assert_running {
     my $self    = shift;
     my $running = shift;
+    my $cmd = '(systemctl is-active named || true) | grep -E';
 
     if ($running) {
-        assert_script_run '(systemctl is-active named || true) | grep -E "^active"', timeout => 300;
+        $cmd .= ' "^active"';
+        if (script_run $cmd) {
+            sleep 15;    # Service may take a bit of time to actually start
+            record_soft_failure 'bsc#1093029';
```

Comment:
Hm, if the service never starts up the same soft-fail would be recorded. I guess it would be better to check for the three different cases:

1. $cmd returns true -> continue further, everything fine
2. $cmd returns false the first time but within a grace period returns true -> record_soft_failure
3. $cmd returns false and never true (e.g. after 7*5s) -> fail

Reply:
Agreed